### ISSUE TITLE
Bug Report: Training Loss Metric Depends on Batch Size

### ISSUE THREAD

**Issue by simon-jonas-buehrer** [ligerlac/torchlogix#15]
## Issue
Training loss metric is incorrectly scaled and depends on batch size, making runs with different batch sizes incomparable.

### Observed Behavior
```
Batch size 32  → train_loss ≈ 9.08
Batch size 128 → train_loss ≈ 0.54  (17x smaller)
```

## Root Cause

Two compounding issues:

1. **Loss accumulation bug** 
https://github.com/ligerlac/torchlogix/blob/0be8eae0d526462c77b617a5da9ebe9642825a4f/experiments/train.py#L341-L342
   Then dividing by `n` introduces batch-size dependency.

2. **Validation loader length multiplier**: 
https://github.com/ligerlac/torchlogix/blob/0be8eae0d526462c77b617a5da9ebe9642825a4f/experiments/train.py#L363

## Solution
Replace with mean-of-batch-means approach (match validation loss computation):

```python
# In initialization
running_train_losses = []

# During training loop
running_train_losses.append(loss.detach().item())

# At evaluation
{"train_loss": np.mean(running_train_losses)}
```

This produces:
- Loss per sample (batch-size independent)
- Comparable with validation loss
- Proper metric for training curve analysis

**Comment by ligerlac** [comment]
#34
